src/Symfony/Component/Security/Http/Authenticator/Passport/PassportInterface.php

```diff
@@ -43,9 +42,7 @@ public function hasBadge(string $badgeFqcn): bool;
     public function getBadge(string $badgeFqcn): ?BadgeInterface;
 
     /**
-     * Checks if all badges are marked as resolved.
-     *
-     * @throws BadCredentialsException when a badge is not marked as resolved
+     * @return array<string, BadgeInterface>
```

Comment:
What does the `string` represent?

That should be documented. 

I suggest changing to `list<BadgeInterface>`

Reply:
make it  `array<class-string<BadgeInterface>, BadgeInterface>` :)

Reply:
I see that the current `PassportTrait` is using a fqcn as index. But that is an implementation detail, right? If so, the contract should be simpler. 

If we really want to use the fact that the keys are a fqcn, we should add that in the contract (interface method doc block).

Reply:
it's not IMHO: `getBadge(string $badgeFqcn)`

Reply:
> we should add that in the contract

yes this is the contract ;)

Reply:
Yeah, Either `array<class-string<BadgeInterface>, BadgeInterface>` if we want the contract to include that info. Or just  `list<BadgeInterface>`. 

I prefer the latter. Especially since we never use the keys.

Reply:
OH, i do agree a list could work. The internal structure is an implementation detail indeed (regardless of how getBadge works).

But im generally fine with the structure at the contract level :+1:

Reply:
> it's not IMHO

It still is =)

That method has nothing to do with how the badges are stored internally. I can imagine at least two implementations where we don't store fqcn as keys but we would still respect this method. 


(There are some serious race conditions in these comments =) )

Reply:
a list implies multiple of the same type, im not sure that should be conveyed.

@wouterj knows best from the design POV :)

Reply:
I'm for making that requirement part of the contract. Updated to `array<class-string<BadgeInterface>, BadgeInterface>` with description. Hope it's ok for everyone

Reply:
if the return type is documented as `list<BadgeInterface>`, the getter must apply `array_values`. The contract of `list` is that keys are sequential integers from 0 to `$count-1`.

Reply:
thanks for confirming